src/bg/execute.js

```diff
@@ -6,7 +6,10 @@ TODO: Make document_start execution time work as intended.
 */
 
 function executeUserscriptOnNavigation(detail) {
-  if (false === getGlobalEnabled()) return;
+  if (false === getGlobalEnabled()
+      || detail.transitionQualifiers.indexOf('forward_back') !== -1) {
```

Comment:
See comment on issue, it's possible for the entire after-scripts-run document to be cached, and it's possible not to.  What if you go forward ten times and then back ten times, what happens?